Automattic/woocommerce-payments
# Add filter to force decimal points in refund form

Fixes #2577

#### Changes proposed in this Pull Request

The decimal places in a price wasn't taken into account <strike>when formatting the price with `wc_format_localized_price`. This adds a filter to the `woocommerce_format_localized_price` hook to ensure that the decimal points of the output matches the store currency.</strike>

The number of decimal places are defined by `WC_ROUNDING_PRECISION` constant when calculating the input value in the refund form here: https://github.com/woocommerce/woocommerce/blob/32993b8aaf36be0c03c5a572e83ee61bb90927e6/assets/js/admin/meta-boxes-order.js#L372-L384

This PR overrides that constant by appending a script to the `<head>` element when MC has _more than one currency available for use_ and the current page is the **WooCommerce > Orders > Order Details** page:

```php
/**
	 * Reduces the client rounding precision to 2
	 *
	 * @return  void
	 */
	public function set_client_rounding_precision() {
		return;
		$screen = get_current_screen();
		if ( 'post' === $screen->base && 'shop_order' === $screen->post_type ) :
			$rounding_precision = wc_get_price_decimals() ?? wc_get_rounding_precision();
			?>
		<script>woocommerce_admin_meta_boxes.rounding_precision = <?php echo intval( $rounding_precision ); ?>;</script>
			<?php
		endif;
	}
``` 

### Before the fix is applied:
![before_fix](https://user-images.githubusercontent.com/3295/129174939-69985016-aca5-4420-9bfe-cd2e80306dff.gif)

### After the fix is applied:
![after_fix](https://user-images.githubusercontent.com/3295/129174975-e59a88a3-7518-4a8c-835c-5926f68ec1ef.gif)










#### Testing instructions





* Go to any order with an item having a price with fractions
* Try to refund the item, on the number input, increase the refund product number to 1
* Check if the price input contains a value with the same decimal places like the price shown above of it.

-------------------
